src/core/handlers/RequestHandler.ts

```diff
@@ -32,7 +32,6 @@ export type ResponseResolverReturnType<
 > =
   | (BodyType extends undefined ? Response : StrictResponse<BodyType>)
   | undefined
```

Comment:
I'm thinking if `undefined` shouldn't be removed as well. The current return type allows for this:

```js
rest.get('/resource', () => {
  // side-effect
})
```

_Edit:_ That use case is still allowed. See [this](https://github.com/mswjs/msw/issues/1207#issuecomment-1536592305) for reference. You can return nothing (but you do have to have a return!) to fall through handlers.